### PR TITLE
ceph-volume: systemd: Give wal/db devices time to be detected

### PR DESCRIPTION
**Pull request by coreycb**
Fixes: https://tracker.ceph.com/issues/40100
Signed-off-by: Corey Bryant <corey.bryant@canonical.com>

### CODE REVIEW

File: src/ceph-volume/ceph_volume/systemd/main.py [hunk 1]
```diff
@@ -50,6 +51,31 @@ def parse_osd_uuid(string):
     return osd_uuid
 
 
+def get_block_volume(string):
+    volumes = Volumes()
+    osd_id = string.split('-', maxsplit=1)[0]
```
Comment (dcasier) [inline src/ceph-volume/ceph_volume/systemd/main.py]: Python 2.x compatibility :
string.split('-', 1)

File: src/ceph-volume/ceph_volume/systemd/main.py [hunk 2]
```diff
@@ -50,6 +51,31 @@ def parse_osd_uuid(string):
...
+    osd_fsid = string.split('-', maxsplit=1)[1]
```
Comment (dcasier) [inline src/ceph-volume/ceph_volume/systemd/main.py]: Python 2.x compatibility :
string.split('-', 1)

File: src/ceph-volume/ceph_volume/systemd/main.py (hunk 1 above)
Comment (coreycb) [reply src/ceph-volume/ceph_volume/systemd/main.py]: good catch, thanks

**Comment by javacruft** [comment]
I think that this feature needs to be pushed down into the LVM activation module; the systemd main module does not really do much other than call lvm trigger, and will retry if it fails, so having the code that waits for LVS's to appear in the LVM module would make more sense IMHO

**Comment by coreycb** [comment]
I think we can abandon this PR in favor of https://github.com/ceph/ceph/pull/28791

**Pull request closed without merging**